test/e2e/run_networking_test.go

```diff
@@ -1097,55 +1097,56 @@ EXPOSE 2004-2005/tcp`, ALPINE)
 	})
 
 	It("podman run check dns", func() {
+		SkipIfCNI(podmanTest)
```

Comment:
This sure was complicated! And it's a big one, so, extra eyes please.

This test has never worked in CNI. It **passed**, but it never **worked**. It passed because `nslookup` in `quay.io/libpod/alpine:latest` checks `/etc/hosts` before calling the resolver. This is arguably a bug, and it has been fixed in later `alpine` images, including the one on which `quay.io/libpod/testimage:20221018` is based.

To the best of my understanding, CNI does not have a DNS server for cotnainers, so this test should be skipped. If my understanding is incorrect, someone please yell loudly... and then that same someone please figure out what is broken.

Reply:
CNI does have a dns support but that a needs podman-plugins (dnsname) installed b) only works for one network per container which is the case here so that should not be the problem.

I took a look at the error and it seems like it does in fact return the proper ip but then asks for more domain which then fail due different search domains.

In any case CNI is deprecated and nobody from us will touch dnsname ever again so skipping SGTM